subprojects/enterprise-operations/src/main/java/org/gradle/api/internal/artifacts/ComponentVariantIdentifier.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts;
+
+import org.gradle.api.internal.artifacts.component.ComponentIdentifier;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Identifier of a variant of a component.
+ */
+public interface ComponentVariantIdentifier {
```

Comment:
🤔 I wonder if we should inline this type. It is nice that we pull things together that belong to one variant, though I am not convinced it is worth it given that we use this only in `TransformationIdentity`.